database/dialect.go

```diff
@@ -22,6 +22,7 @@ const (
 	DialectTiDB       Dialect = "tidb"
 	DialectVertica    Dialect = "vertica"
 	DialectYdB        Dialect = "ydb"
+	DialectTurso      Dialect = "turso"
```

Comment:
I'm not sure we need a new exported dialect in the core package since the driver is compatible with the existing `DialectSQLite3` dialect?

Reply:
Turso is backward compatible with SQLite. But it does have support for SQL queries which are not compatible with SQLite. 

Second reason is because of user experience. I thought it is better to have uniform driver/dialect usage, instead of mixed Turso/SQLite

Lastly, if we add any dialect changes tomorrow, user's don't have to update their code. All changes would be within goose.

Reply:
Fair.